### ISSUE TITLE
Dalaran-WoW Hunter Auto Shot Bug

### ISSUE THREAD

**Issue by hrtyv** [dalaranwow/dalaran-wow#4653]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)

Hunter´s Auto Shot keep stopping when - Knockdown - Feared -  other imobilized actions happens to the Hunter.

It can be force started by using Feign Death or untarget and retarget the enemy followed by rightclick, at the moment.

**Expected behaviour**: Tell us what should happen instead

Auto Shot should be starting from it self when imobilized action ends!
Or  you should atleast be able to either right click target to start Auto Shot again or force it by using Macro like: 

#showtooltip Arcane Shot
/Cast! Auto Shot
/Cast Arcane Shot

**Steps to reproduce the problem**:
In the test below - i am only using right click to start Auto Shot / input for Feign Death / input for Arcane Shot macro mentioned earlier.
1. 
[With Feign Death.zip](https://github.com/dalaranwow/dalaran-wow/files/888090/With.Feign.Death.zip)
2. 
[Out Feign Death.zip](https://github.com/dalaranwow/dalaran-wow/files/888091/Out.Feign.Death.zip)
3. 
[Feign Death short.zip](https://github.com/dalaranwow/dalaran-wow/files/888092/Feign.Death.short.zip)

**Include proofs for this behaviour**

https://www.youtube.com/watch?v=xBootz8pwL4   
-  In the video im in a duel with a Warlock that uses fear on me, i am trying to Auto Shot him when fear ends i am right clicking him trying to start Auto Shot again, because it does not start  by itself. You can see when i am using Feign Death and right click the Auto Shot will finally start.
I am also using Arcane Shot with /Cast! Auto shot Macro to proove that it does not work either. Same counts for all other shot Macro´s.

Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

**Auto Shot:**
- spell=75

**Feign Death**
- spell=5384

**Disablers that i know of:**
(There is probably more)

**Gormok the Impaler**
- npc=35469 - spell=66330/staggering-stomp

**Onyxia**
- npc=10184 - spell=18431/bellowing-roar

**Warlocks** 
- spell=65809/fear - spell=17928/howl-of-terror - spell=68141/death-coil

**Archavon-the-stone-watcher**
- npc=31125 - spell=58663/stomp

**Kologarn**
- npc=32930 - spell=63783/shockwave

You can research this on several pages like http://wotlk.openwow.com or http://wowhead.com/

**Include Screenshots from the issue if necessary**

There is a video and problem records with screens.

Hope this will be enough to help you guys solve the Hunters issues because the Auto Shot is a BIG part of Hunters Damage.

**_Thanks in advance_**

**Comment by TeridanDW** [comment]
Hello, thank you for the report @hrtyv .
Issue related to #834 and maybe #3740

**Comment by hrtyv** [comment]
Thanks for the reply. 

I hope that this issue and the things I implemented can be of any help to the developers and that the problem will be solved when the ICC patch gonna launch, hopefully sooner than that.

Thanks

**Comment by Max0r847** [comment]
Wow this is huge. Very annoying

**Comment by Rushor** [comment]
Already reported in https://github.com/dalaranwow/dalaran-wow/issues/834. Most of the issues concerning AutoRepeatSpells were already solved there in ttps://github.com/dalaranwow/dalaran-wow/issues/834. 

Once ttps://github.com/dalaranwow/dalaran-wow/issues/834 was inspected, a tester can add if the commit also fixes the autoshot cast after the immobilized action ends.

Duplicate for now.
